snuba/query/snql/parser.py

```diff
@@ -632,11 +639,65 @@ def parse(exp: Expression) -> Expression:
     query.transform_expressions(parse)
 
 
+ARRAY_JOIN_MATCH = FunctionCallMatch(
+    StringMatch("arrayJoinExplode"),
```

Comment:
I think this should be named differently as it is not an arrayJoin.
We have made some effort to make it clear to the product teams that arrayJoins are essentially self joins over the table thus they expand the size of the data set to query. This has a major performance impact and it should be dealt with carefully.

This is not an arrayJoin as it does not expand the size of the dataset to query, it is only an array operation and does not brings those performance issues.

Since this is a shortcut to arrayAll and arrayExists, and since the user cannot use arrayAll and arrayExists as we do not have lambdas in the language. Why not calling them as such?

`arrayAll(col, op, val)`
`arrayExist(col, op, val)`

And let the client pick (as they would do with clickhouse directly). It feels it would also have a more intuitive behavior for the user  instead of silently relying on a distinction between positive and negative operators

Reply:
I like this better, I changed it.